compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -158,7 +159,8 @@ class ScenarioService extends ScenarioServiceGrpc.ScenarioServiceImplBase {
       respObs: StreamObserver[GCContextsResponse],
   ): Unit = {
     val ctxIds = req.getContextIdsList.asScala.toSet
-    contexts = contexts.filterKeys(ctxId => ctxIds.contains(ctxId));
+    val ctxToRemove = contexts.keySet.filter(ctxId => !ctxIds.contains(ctxId))
```

Comment:
Why not [`Set.diff`](https://www.scala-lang.org/api/2.12.3/scala/collection/Set.html#diff(that:scala.collection.GenSet[A]):This)?

Reply:
that’s the wrong way around, I want to keep the elements in `ctxIds`. `contexts.keySet.diff(ctxIds)` would remove them.

Reply:
nvm, you are completely right. I’ve switched to `Set.diff`.